### DISCUSSION TITLE
Nova licença do React. Mudou pra melhor ou pior?

### DISCUSSION THREAD

**Discussion by derekstavis** [frontendbr/forum#742] [Discussão]
Olá pessoal,

Como muitos sabem, o relicenciamento ocorreu devido aos usuários e especuladores estarem questionando o modelo de licenciamento anterior, que era baseado em uma licença BSD de 3 cláusulas e um grant adicional de patentes, que garantia direitos de uso de possíveis patentes do Facebook relacionadas ao pacote licenciado. O problema é que caso o usuário da biblioteca processasse o Facebook por algum motivo, esse grant adicional de patentes seria revogado.

Com o feedback ostensivo da internet, o Facebook resolveu relicenciar o pacote na licença mais simples possível: A licença MIT.

Porém pra mim o problema é o seguinte:

**ANTES**

https://github.com/facebook/react/blob/v15.5.4/LICENSE
https://github.com/facebook/react/blob/v15.5.4/PATENTS

**DEPOIS**

https://github.com/facebook/react/blob/master/LICENSE
https://github.com/facebook/react/blob/master/PATENTS

Sim, tem um 404 mesmo no PATENTS do DEPOIS. Fico na duvida se a mudança foi pra melhor ou pra pior. O que poderá ocorrer agora caso o Facebook patenteie alguma tecnologia do React? Antes era super claro que eu tinha direito de usar, mas agora não mais! Qual a vossa opinião?

**Comment by fdaciuk** [comment]
Acho que se for seguir essa linha de raciocínio, teríamos problemas em usar qualquer lib com a licença MIT, que tenha uma grande empresa por trás, correto?

**Comment by matheusml** [comment]
Oi @derekstavis,

Se desse pra resumir a licença MIT em um _tweet_, eu o faria assim: 

> faça o que você quiser com isso, só não me perturbe

Lembrando que Vue, Ember, Angular e praticamente quase todos concorrentes também estão usando a licença MIT, então se por algum motivo aleatório você não gosta muito dela, não vai ter pra onde correr.

**Comment by felipefialho** [comment]
MIT é a forma mais simples de licença. Não da dor de cabeça para quem usa e nem para o dono da lib.

Pra mim mudaram pra melhor.

**Comment by danilovaz** [comment]
@derekstavis O Rails é MIT a não sei quantos anos, com a empresa do DHH por trás e até onde eu sei, sem nenhum problema com isso. E como o @matheusml disse, todos os outros fw JS são MIT também. Então acredito que mudaram para melhor.

**Comment by dvmuccillo** [comment]
Tava lendo hoje no Medium a sequência  de acontecimentos que gerou a mudança, para quem considerar relevante: “Facebook just changed the license on React. Here’s a 2-minute explanation why.” @ossia https://medium.freecodecamp.org/facebook-just-changed-the-license-on-react-heres-a-2-minute-explanation-why-5878478913b2

**Comment by didiraja** [comment]
Existe algum texto explicativo iniciante falando sobre esses diferentes tipos de licença?

**Comment by fdaciuk** [comment]
Apesar da licença antiga e da nova licença do React não mudarem **em nada** a vida de praticamente ninguém, foi bom ter acontecido porque agora a galera começa a se ligar mais nessa questão de licença, que já existe a mil anos, mas normalmente não nos preocupamos com isso, só usamos o produto sem nem ler seu contrato. Ponto positivo :)

@didiraja dá uma [olhada aqui](https://www.slideshare.net/ShijieFeng/open-source-software-licenses-for-humans), pode ajudar :)

**Comment by felipechalreo** [comment]
Posso estar enganado mas acho que o Facebook so mudou de idea por causa do Wordpress ( https://ma.tt/2017/09/on-react-and-wordpress/ )
